rampwf/utils/testing.py

```diff
@@ -590,42 +697,89 @@ def assert_submission(ramp_kit_dir='.', ramp_data_dir='.',
+    foldwise_best_predictions_test_list = []
+    for fold_i, valid_is in enumerate(valid_is_list):
+        _print_title('CV fold {}'.format(fold_i))
+        ground_truths_valid = problem.Predictions(y_true=y_train[valid_is])
+        predictions_valid_list = []
+        predictions_test_list = []
+        for submission in submissions:
+            module_path = join(ramp_kit_dir, 'submissions', submission)
+            training_output_path = join(module_path, 'training_output')
+            fold_output_path = join(
+                training_output_path, 'fold_{}'.format(fold_i))
+            y_pred_train = _load_y_pred(
+                problem, data_path=ramp_data_dir,
+                input_path=fold_output_path, suffix='train')
+            y_pred_test = _load_y_pred(
+                problem, data_path=ramp_data_dir,
+                input_path=fold_output_path, suffix='test')
+            predictions_valid = problem.Predictions(
+                y_pred=y_pred_train[valid_is])
+            predictions_valid_list.append(predictions_valid)
+            predictions_test = problem.Predictions(y_pred=y_pred_test)
+            predictions_test_list.append(predictions_test)
+
+        best_index_list = blend_on_fold(
+            predictions_valid_list, ground_truths_valid, score_types[0],
+            min_improvement=min_improvement)
+
+        # we share a unit of 1. among the contributive submissions
+        unit_contributivity = 1. / len(best_index_list)
+        for i in best_index_list:
+            contributivitys[i] += unit_contributivity
+
+        combined_predictions_valid_list.append(
+            problem.Predictions.combine(predictions_valid_list))
+        foldwise_best_predictions_valid_list.append(predictions_valid_list[0])
+        combined_predictions_test_list.append(
+            problem.Predictions.combine(predictions_test_list))
+        foldwise_best_predictions_test_list.append(predictions_test_list[0])
+
+    contributivitys /= len(cv)
+    contributivitys_df = pd.DataFrame()
+    contributivitys_df['submission'] = np.array(submissions)
+    contributivitys_df['contributivity'] = np.round(contributivitys, 3)
+    contributivitys_df = contributivitys_df.reset_index()
+    contributivitys_df = contributivitys_df.sort_values(
+        'contributivity', ascending=False)
+    print(contributivitys_df.to_string(index=False))
+
+    training_output_path = join(ramp_kit_dir, 'training_output')
+    if not os.path.exists(training_output_path):
+        os.mkdir(training_output_path)
+    # bagging the foldwise ensembles
+    _bag_submissions(
+        problem, cv, y_train, y_test, combined_predictions_valid_list,
+        combined_predictions_test_list, training_output_path,
+        ramp_data_dir=ramp_data_dir, score_type_index=0,
+        save_y_preds=save_y_preds, score_table_title='Combined bagged scores',
+        score_f_name_prefix='foldwise_best')
+    # bagging the foldwise best submissions
```

Comment:
To what extent is this useful? Because this is not based on the blending?

Reply:
No, it's an alternative way to combine folds. Most of the time the combined combined is better, but I like to look at the combined foldwise best.